### PR TITLE
Feature/add items

### PR DESCRIPTION
**Pull request by NeeleWaehnke**
- Create 'Add'Button 
- Add function to button
-Basic styling (not finished) for form, buttons and footer section

### CODE REVIEW

File: components/AddButton.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import styled from 'styled-components';
+
+export default function AddButton({ setAddActive }) {
+  return (
+    <>
+      <StyledButton
+        onClick={() => {
+          setAddActive(true);
+          console.log(true);
+        }}
```
Comment (MerleMak) [inline components/AddButton.js:11]:
```suggestion
        onClick={onClick}
```

I would pass the information as an onClick until you reach the component where you actually wan to work with the onClick event :)

Also: always delete your console logs before commiting :)

File: components/AddForm.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
+import CheckIcon from '../public/assets/check.svg';
+import styled from 'styled-components';
+
+export default function AddForm({ onAddItem, setAddActive }) {
+  function handleSubmit(event) {
+    event.preventDefault();
+    const form = event.target.elements;
+    const data = {
```
Comment (MerleMak) [inline components/AddForm.js:8]:
```suggestion
    const newItem = {
```
Better to rename this constant as "data" is very generic. Try and choose names that will help to explain what is happening :)

File: components/AddForm.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      name: form.name.value,
+      date: form.date.value,
+      quantity: form.quantity.value,
+    };
+    onAddItem(data);
```
Comment (MerleMak) [inline components/AddForm.js:13]:
```suggestion
    onAddItem(newCard);
```

File: components/Footer.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import AddForm from './AddForm';
+import { useState } from 'react';
+import styled from 'styled-components';
+
+export default function Footer({ onAddItem }) {
+  const [addActive, setAddActive] = useState(false);
+  return (
+    <StyledSection>
+      {addActive === true ? (
+        <AddForm onAddItem={onAddItem} setAddActive={setAddActive} />
```
Comment (MerleMak) [inline components/Footer.js:11]:
```suggestion
        <AddForm onAddItem={onAddItem} onClick={setAddActive} />
```
See my comment in the AddButton component :)

File: components/Footer.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import AddButton from './AddButton';
+import AddForm from './AddForm';
+import { useState } from 'react';
+import styled from 'styled-components';
+
+export default function Footer({ onAddItem }) {
```
Comment (MerleMak) [inline components/Footer.js:6]: I don't fully understand why you are using a Footer as the container for your Form. In my opinion you could delete the Footer and directly add the Addform to your index.js 
A Footer is usually used for the impressum, links to social media or similar, some devs like to add their navbar to the footer, but you don't necessarily need a footer for your app. 
So you could transfer all this code, including the state and the conditional rendering into your AddForm component. This way you could avoid this extra layer and passing your state up and down :)

File: components/AddForm.js [hunk 6]
```diff
@@ -0,0 +1,70 @@
+import CheckIcon from '../public/assets/check.svg';
+import styled from 'styled-components';
+
+export default function AddForm({ onAddItem, setAddActive }) {
```
Comment (MerleMak) [inline components/AddForm.js:4]: See my remark in the Footer component, you could avoid passing the state here, by just maintaining the state here. 
Another option is to maintain in the index.js, then you would still need to pass the state as you are doing here :), i would prefer the first option though :)

File: pages/index.js [hunk 7]
```diff
@@ -1,11 +1,22 @@
 import ItemList from '../components/ItemList';
 import Header from '../components/Header';
+import Footer from '../components/Footer';
+import { useState } from 'react';
+import { nanoid } from 'nanoid';
+import { dummyitems } from '../lib/data';
 
 export default function Home() {
+  const [items, setItems] = useState([...dummyitems]);
+  function handleAddItem(newItem) {
```
Comment (MerleMak) [inline pages/index.js:10]: Try and name your "brokkolis" the same when possible. You named the argument you passed in AddForm "data", I suggest you name both newItem :)

File: components/AddForm.js [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+  function handleSubmit(event) {
+    event.preventDefault();
+    const form = event.target.elements;
+    const data = {
+      name: form.name.value,
+      date: form.date.value,
+      quantity: form.quantity.value,
```
Comment (MerleMak) [inline components/AddForm.js:11]:
```suggestion
      quantity: form.quantity.value,
      id: nanoid(),
```

Its probably best to already add an ID here, for example with nano ID, this could save you some trouble later :)

File: pages/index.js [hunk 9]
```diff
@@ -1,11 +1,22 @@
...
+    setItems([{ name, date, quantity, id: nanoid(), ...newItem }, ...items]);
```
Comment (MerleMak) [inline pages/index.js:11]:
```suggestion
    setItems([newItem, ...items]);
```

Oh now i see that you added an ID already!
I think its is a little bit more structured if you add the id where you build the whole object (in AddForm.js), but as you wish. this comment is only a suggestion :)

**Review by MerleMak — COMMENTED** [review]
Great Job Neele!
I left some remarks for you, 
especially consider if you want to keep the footer as an extra component, as this adds some complexity which is not yet needed.
Other than that, your code looks great, its very good to understand!
Keep it up like that 💪🏼👍🏼

File: next.config.js [hunk 10]
```diff
@@ -13,6 +13,14 @@ const nextConfig = {
...
     ],
   },
+  webpack(config) {
+    config.module.rules.push({
+      test: /\.svg$/i,
+      issuer: /\.[jt]sx?$/,
+      use: ['@svgr/webpack'],
+    });
+    return config;
+  },
```
Comment (onebarloop) [inline next.config.js:23]: Why did you need to configure webpack? I'm really interested 😁

**Comment by onebarloop** [comment]
Great work Neele! :)
